target/linux/ramips/dts/mt7621_tplink_er605-v2.dts

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "TP-Link ER605 v2";
+	compatible = "tplink,er605-v2", "mediatek,mt7621-soc";
+
+	chosen {
+		// FIXME: root is hardcoded here and I do not know what this was on a 'real' device
```

Comment:
From the stock dtb:
```
        chosen {
                bootargs = "console=ttyS0,115200 noinitrd";
        };
```
So just `console=ttyS0,115200` should suffice here I believe.

Reply:
Unfortunately without that things do not currently boot.  I get this:
```
[    2.870727] /dev/root: Can't open blockdev
[    2.874824] VFS: Cannot open root device "(null)" or unknown-block(0,0): error -6
[    2.882299] Please append a correct "root=" boot option; here are the available partitions:
```

As far as I can see there is no kernel cmd line being passed.
I'm not sure why this would be happening though - I'll have to look more into this

Reply:
I believe you also require the correct compatible labelling for the "firmware" partition. `compatible = "denx,uimage";` or `compatible = "denx,fit";` etc.

Reply:
I have removed the `root` bits from `bootargs` - so now root volume should be passed by the bootloader as with the oem firmware.


> I believe you also require the correct compatible labelling for the "firmware" partition. `compatible = "denx,uimage";` or `compatible = "denx,fit";` etc.

Could you please clarify this a little bit?
`firmware` partition contains ubi volumes - and only one of them is the kernel volume... Do I still need these bits given this configuration?

Thanks!

Reply:
Is it booting up with the current changes?
If it is booting up you can also remove the FIXME on top of this line.

Reply:
Yes, sorry.
Yes it boots, at least with initramfs.
I've removed the comment.

Thanks!